blockchain/chain_sync/src/sync.rs

```diff
@@ -188,15 +230,82 @@ where
         let best_weight = heaviest_tipset.blocks()[0].weight();
         let target_weight = fts.blocks()[0].header().weight();
 
-        if !target_weight.lt(&best_weight) {
-            // Store incoming block header
-            self.chain_store.persist_headers(&fts.tipset()?)?;
+        if target_weight.gt(&best_weight) {
             // Set peer head
-            self.sync_manager.set_peer_head(from, fts.tipset()?);
+            self.set_peer_head(from, fts.tipset()?.clone()).await?;
         }
         // incoming tipset from miners does not appear to be better than our best chain, ignoring for now
         Ok(())
     }
+    pub async fn set_peer_head(&mut self, peer: &PeerId, ts: Tipset) -> Result<(), Error> {
+        // update peer heads map
+        self.peer_manager
+            .insert_peer_head(peer.clone(), ts.clone())
+            .await;
+
+        // initial sync
+        if self.get_state() == &SyncState::Init {
+            if let Some(best_target) = self.select_sync_target().await {
+                self.sync(best_target).await?;
+                return Ok(());
+            }
+        }
+        self.schedule_tipset(ts.clone()).await?;
```

Comment:
Not used after, can just move to remove unnecessary clone
```suggestion
        self.schedule_tipset(ts).await?;
```